Switch to nullptr everywhere.

clalancette opened the pull request:
Signed-off-by: Chris Lalancette <clalancette@openrobotics.org>

This is a trivial patch to just replace `NULL` -> `nullptr` everywhere.  Just something I noticed while looking into other things here.

clalancette committed:
Switch to nullptr everywhere.

Signed-off-by: Chris Lalancette <clalancette@openrobotics.org>

allenh1 commented on rviz_common/src/rviz_common/config.cpp:
```diff
@@ -282,14 +282,14 @@ bool Config::mapGetString(const QString & key, QString * value_out) const
 void Config::makeValid()
 {
-  if (node_.get() == NULL) {
+  if (node_.get() == nullptr) {
```
@clalancette Any reason not to change it like this? (since you're here and all)

```suggestion
  if (!node_) {
```

clalancette replied:
Mostly this was a search-n-replace, so I didn't consider it.

I personally prefer the more explicit code for readability.  It's really easy for me to miss the `!` when glancing over code, while the `== nullptr` stands out to me.  But if others feel strongly about it, I can change it.

allenh1 replied:
I don't mind the `nullptr`, but the `.get()` feels kinda silly.

Up to you though, it's really not important in any sort of capacity.

jacobperron replied:
+1 to remove the `.get()` (which doesn't seem necessary).

clalancette replied:
All right, I've gone ahead and removed the `.get()`.  I'll run CI now.

jacobperron reviewed (commented):
LGTM, just one comment about dropping the `.get()` where it shows up.

clalancette commented:
* Linux [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_linux&build=11985)](http://ci.ros2.org/job/ci_linux/11985/)
* Linux-aarch64 [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_linux-aarch64&build=7017)](http://ci.ros2.org/job/ci_linux-aarch64/7017/)
* macOS [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_osx&build=9769)](http://ci.ros2.org/job/ci_osx/9769/)
* Windows [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_windows&build=11924)](http://ci.ros2.org/job/ci_windows/11924/)

jacobperron commented:
Looks like there's some test regressions on Windows :thinking:

clalancette commented:
> Looks like there's some test regressions on Windows thinking

It's unlikely.  I think these are the same CI problems that have been plaguing us for a while on this repository.  Most likely if we were to merge this, nightly builds would be green.

That being said, I'll spend a bit of time and see if I can figure out what is going on here (I wasn't able to reproduce it on a local Windows VM the last time I tried).

jacobperron commented:
Oh, why do they not show up in the nightlies? Are they known flakes?

clalancette commented:
> Oh, why do they not show up in the nightlies? Are they known flakes?

Oddly, they aren't flaky at all; they always succeed in the nightlies, and they always fail in CI.  I just have no idea why :).

jacobperron commented:
Recent Windows CI builds for rviz appear to have been passing:

- https://ci.ros2.org/job/ci_windows/11915/
- https://ci.ros2.org/job/ci_windows/11917/

clalancette commented:
> Recent Windows CI builds for rviz appear to have been passing:
> 
>     * https://ci.ros2.org/job/ci_windows/11915/
> 
>     * https://ci.ros2.org/job/ci_windows/11917/

Hm, so they are.  Let me try to match those settings and see what happens:

* Windows [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_windows&build=11926)](http://ci.ros2.org/job/ci_windows/11926/)

jacobperron commented:
Strange, I don't know why the difference in package selection args would affect the outcome of the tests (it looks like tests were run in both jobs).

clalancette commented:
> Hm, so they are.  Let me try to match those settings and see what happens:

So, weird.  The latest Windows build and test worked just fine.  The difference to the previous one is that I changed the build line from `--packages-up-to rviz_common rviz_rendering` to `--packages-up-to rviz2`, and I changed the test line from `--packages-select rviz_common rviz_rendering` to `--packages-select regex rviz`.  I don't understand why that would make a difference, but that does explain why we don't see it in nightlies.

In any case, I think it is safe to say it is not part of this PR, so an approval would be appreciated :).

clalancette commented:
Thanks for the review!  Merging.
